src/backend/metal/src/command.rs

```diff
@@ -214,27 +220,64 @@ impl command::Buffer<Resources> for CommandBuffer {
         }
 
         if let Some(depth) = targets.depth {
+            let stencil_attachment = if let Some(stencil) = targets.stencil {
+                debug_assert!(depth == stencil, "Depth and stencil targets must be the same if both are present");
+                Some(render_pass_descriptor.stencil_attachment())
+            } else {
+                None
+            };
+
             let attachment = render_pass_descriptor.depth_attachment();
             attachment.set_texture(unsafe { *(depth.0) });
+            if let Some(stencil_attachment) = stencil_attachment {
+                stencil_attachment.set_texture(unsafe { *(depth.0) });
+            }
 
             if let Some(layer) = depth.1 {
                 attachment.set_slice(layer as u64);
+                if let Some(stencil_attachment) = stencil_attachment {
+                    stencil_attachment.set_slice(layer as u64);
+                }
             }
 
             // do we need to handle any other cases?
             attachment.set_store_action(MTLStoreAction::Store);
+            if let Some(stencil_attachment) = stencil_attachment {
+                stencil_attachment.set_store_action(MTLStoreAction::Store);
+            }
 
-            let clear = unsafe { *depth.2 };
-
-            if let Some(val) = clear {
-                attachment.set_load_action(MTLLoadAction::Clear);
-                attachment.set_clear_depth(val as f64);
+            if let Entry::Occupied(mut clear_entry) = self.dsv_clear.entry(depth) {
```

Comment:
The reason I used it is that the entry may need to be deleted entirely, not just modified. If get_mut is used there would be two lookups in that case. For the depth target this is conditional so I can't simplify it to a remove(), but the render target analogue always removes the entry if present. I'll change that to remove() instead of mucking about with Entries.